src/controllers/admin/mentor.controller.ts

```diff
@@ -105,3 +106,33 @@ export const getAllMentorEmails = async (
     throw err
   }
 }
+
+export const updateMentorAvailability = async (
+  req: Request,
+  res: Response
+): Promise<ApiResponse<Mentor>> => {
+  try {
+    const user = req.user as Profile
+    const { availability } = req.body
+    const { mentorId } = req.params
+
+    if (user.type !== ProfileTypes.ADMIN) {
+      return res.status(403).json({ message: 'Only Admins are allowed' })
+    }
+
+    const { mentor, statusCode, message } = await updateAvailability(
```

Comment:
Here we can use the existing service `updateAvailability` in `src/services/mentor.service.ts`

Reply:
But exisiting function require user profile as paramter 
```
export const updateAvailability = async (
  user: Profile,
  availability: boolean
): Promise<{ statusCode: number; updatedMentorApplication: Mentor }> => {
```. Here we have mentor Id only

Reply:
can you update that(updateAvailability) to be the mentorId as well? Both are the same. @sathudeva7